File: game_bizModule/src/main/java/com/game/bizModule/chat/msg/CGWorldChat.java
package com.game.bizModule.chat.msg;

import com.game.bizModule.chat.bizServ.ITypist;
import com.game.bizModule.chat.bizServ.WorldChatTypist;
import com.game.bizModule.global.AllMsgSerialUId;
import com.game.gameServer.msg.MsgTypeEnum;

/**
 * 世界聊天
 *
 * @author hjj2017
 * @since 2015/8/15
 *
 */
public class CGWorldChat extends AbstractCGChat {
    /**
     * 类默认构造器
     *
     */
    public CGWorldChat() {
        super();
    }

    /**
     * 类参数构造器
     *
     * @param text
     *
     */
    public CGWorldChat(String text) {
        super(text);
    }

    @Override
    public short getSerialUId() {
        return AllMsgSerialUId.CG_WORLD_CHAT;
    }

    @Override
    public ITypist<CGWorldChat> newTypist() {
        return WorldChatTypist.OBJ;
    }

    @Override
    public MsgTypeEnum getMsgTypeInTyping() {
        return MsgTypeEnum.game;
    }
}

File: client_wpf/Xgame.GameBizModule/Chat/Msg/CGWorldChat.cs
using System;

using Xgame.GameClient.Global;
using Xgame.GameClient.Msg;
using Xgame.GamePart.Msg.Type;

namespace Xgame.GameClient.Chat.Msg
{
    /// <summary>
    /// 世界聊天 CG 消息
    /// </summary>
    public partial class CGWorldChat : BaseCGMsg
    {
        /** 聊天内容 */
        public MsgStr _text;

        // @Override
        public override short MsgSerialUId
        {
            get
            {
                return AllMsgSerialUId.CG_WORLD_CHAT;
            }
        }
    }
}
